src/transformStateWithClones.js

```diff
@@ -7,7 +7,41 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClones = [];
+
+  for (const action of actions) {
+    const currentState = {};
+
+    if (stateClones.length !== 0) {
```

Comment:
The condition 'if (stateClones.length !== 0)' is not necessary. You can directly use Object.assign() on an empty object and it will just return an empty object back if there is nothing to copy.